src/i965_avc_encoder_common.h

```diff
@@ -0,0 +1,321 @@
+    struct i965_gpe_resource res_brc_mbenc_curbe_read_buffer;
+    struct i965_gpe_resource res_brc_mbenc_curbe_write_buffer;
+    struct i965_gpe_resource res_brc_const_data_buffer;
+    //brc and mbbrc
+    struct i965_gpe_resource res_mb_status_buffer;
+    //mbbrc
+    struct i965_gpe_resource res_mbbrc_mb_qp_data_surface;
+    struct i965_gpe_resource res_mbbrc_roi_surface;
+    struct i965_gpe_resource res_mbbrc_const_data_buffer;
+
+    //mbenc
+    struct i965_gpe_resource res_mbenc_slice_map_surface;
+
+    //scaling flatness check surface
+    struct i965_gpe_resource res_flatness_check_surface;
+    //me
+    struct i965_gpe_resource s4x_memv_min_distortion_brc_buffer;
+    struct i965_gpe_resource s4x_memv_distortion_buffer;
+    struct i965_gpe_resource s4x_memv_data_buffer;
+    struct i965_gpe_resource s16x_memv_data_buffer;
+    struct i965_gpe_resource s32x_memv_data_buffer;
+
+
+    struct i965_gpe_resource res_image_state_batch_buffer_2nd_level;
+    struct intel_batchbuffer *pres_slice_batch_buffer_2nd_level;
+    // mb code/data or indrirect mv data, define in private avc surface
+
+    //sfd
+    struct i965_gpe_resource res_sfd_output_buffer;
+    struct i965_gpe_resource res_sfd_cost_table_p_frame_buffer;
+    struct i965_gpe_resource res_sfd_cost_table_b_frame_buffer;
+
+    //external mb qp data,application input
+    struct i965_gpe_resource res_mb_qp_data_surface;
+
+    struct i965_gpe_resource res_mad_data_buffer;
+
+    //wp
+    VASurfaceID wp_output_pic_select_surface_id[2];
+    struct object_surface *wp_output_pic_select_surface_obj[2];
+    struct i965_gpe_resource res_wp_output_pic_select_surface_list[2];
+
+    //mb disable skip
+    struct i965_gpe_resource res_mb_disable_skip_map_surface;
+
+    /* PAK resource */
+    //internal
+    struct i965_gpe_resource res_intra_row_store_scratch_buffer;
+    struct i965_gpe_resource res_deblocking_filter_row_store_scratch_buffer;
+    struct i965_gpe_resource res_deblocking_filter_tile_col_buffer;
+    struct i965_gpe_resource res_bsd_mpc_row_store_scratch_buffer;
+    struct i965_gpe_resource res_mfc_indirect_bse_object;
+    struct i965_gpe_resource res_pak_mb_status_buffer;
+    struct i965_gpe_resource res_direct_mv_buffersr[NUM_MFC_AVC_DMV_BUFFERS];//INTERNAL: 0-31 as input,32 and 33 as output
+
+    //output
+    struct i965_gpe_resource res_post_deblocking_output;
+    struct i965_gpe_resource res_pre_deblocking_output;
+
+    //ref list
+    struct i965_gpe_resource list_reference_res[MAX_MFC_AVC_REFERENCE_SURFACES];
+
+    // kernel context
+    struct gen_avc_scaling_context  context_scaling;
+    struct gen_avc_me_context  context_me;
+    struct gen_avc_brc_context  context_brc;
+    struct gen_avc_mbenc_context  context_mbenc;
+    struct gen_avc_wp_context  context_wp;
+    struct gen_avc_sfd_context  context_sfd;
+
+    struct encoder_status_buffer_internal status_buffer;
+
+};
+
+#define MAX_AVC_SLICE_NUM 256
+struct avc_enc_state {
+
```

Comment:
it is common for AVC on GenX. I am thinking about supporting AVC on more platform, such as KBL/BDW etc and move this structure into the common.